### PR TITLE
added isService to bc

### PR DESCRIPTION
**Pull request by jayjaytrn**

### CODE REVIEW

File: core/base_contract.go [hunk 1]
```diff
@@ -275,6 +276,16 @@ func (bc *BaseContract) TracingHandler() *telemetry.TracingHandler {
 	return bc.tracingHandler
 }
 
+// SetIsService sets base contract isService
+func (bc *BaseContract) SetIsService() {
```
Comment (pfi79) [inline core/base_contract.go:280]: Maybe not an exportable method?

File: core/base_contract.go [hunk 2]
```diff
@@ -346,4 +357,6 @@ type BaseContractInterface interface { //nolint:interfacebloat
 
 	setTracingHandler(th *telemetry.TracingHandler)
 	TracingHandler() *telemetry.TracingHandler
+
```
Comment (pfi79) [inline core/base_contract.go:360]: Add the installation method here.

File: core/contract/service.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package contract
+
+type Service interface {
+	SetIsService()
+}
```
Comment (pfi79) [inline core/contract/service.go:5]: delete that.

Comment (jayjaytrn) [reply core/contract/service.go:5]: removed

File: core/base_contract.go (hunk 1 above)
Comment (jayjaytrn) [reply core/base_contract.go:280]: done

**Review by pfi79 — APPROVED** [review]

**Comment by pfi79** [comment]
@jayjaytrn Thank you. That's great.

**Pull request merged**